Enable PInvoke stub compilation for R2R

clamp03 opened the pull request.

clamp03 commented:
I fixed to enable IL STUB compilation in crossgen for ARM target on ARM device.
I think that test failed because this patch generates different files on ARM and X64.

jkotas commented:
> I think that test failed because this patch generates different files on ARM and X64.

Yes, that's the problem. Do you plan to do something about it?

clamp03 committed:
Enable PInvoke stub compilation for R2R

R2R doesn't generate PInvoke stub for arm because of crossbitness issue.
This patch enables PInvoke stub compilation for R2R,
and updates native size, field offset, and so on based on target machine's bitness.

clamp03 commented:
@jkotas Sorry for the late response. I couldn't work for weeks because of COVID19.
I want to make a commit that generates STUB in case of cross bitness next week.
Thank you.

clamp03 commented:
I think a failure on Windows_NTx86 is not by this patch. I will check a failure on Linux arm64.
Thank you.

clamp03 commented:
@jkotas  I think test failures are not by this patch. I don't know how to manage issues. Could you help?

jkotas commented on src/coreclr/src/vm/method.cpp:
```diff
@@ -1329,11 +1329,11 @@ ReturnKind MethodDesc::ParseReturnKindFromSig(INDEBUG(bool supportStringConstruc
 ReturnKind MethodDesc::GetReturnKind(INDEBUG(bool supportStringConstructors))
 {
-#ifdef HOST_64BIT
+#ifdef TARGET_64BIT
```
This ifdef can be just delete (keep the comment though).

clamp03 replied:
Fixed. Thank you.

jkotas commented:
Thanks!

clamp03 commented:
I squashed and made to one commit. And all checkes are passed.
Could you please review or merge? Thank you.

clamp03 commented:
@jkotas Thank you!
